lib/templatesimplifier.cpp

```diff
@@ -1714,7 +1714,7 @@ void TemplateSimplifier::expandTemplate(
                             }
                             // check if type is instantiated
                             for (const auto & inst : mTemplateInstantiations) {
-                                if (Token::simpleMatch(inst.token(), name.c_str())) {
+                                if (Token::simpleMatch(inst.token(), name.c_str(), name.size())) {
```

Comment:
Is it possible to do an overloaded function which takes a `const std::string&` to put all the `c_str()` and `size()` calls in  one place?

Reply:
I was planning to do that. Using `std::string` is discouraged though since the matchcompiler cannot handle it.

Reply:
I would like to keep it like this for now and clean this up at a later date. Maybe this different signature is even something to use in the matchercompiler.